### PR TITLE
[SDP-1118] Hotfix: sanitize disbursement instructions

### PR DESCRIPTION
**Pull request by marwen-abid**
### What
* Trim spaces for all fields in disbursement instructions

### Why
Resolve https://stellarorg.atlassian.net/browse/SDP-1118
Phone number with extra spaces especially, may result in creation of duplicate receivers. 

### Known limitations

[TODO or N/A]

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

**Comment by marwen-abid** [comment]
Tested triggering two disbursements, one having extra spaces in instructions and one without. Only one receiver was created. 
Screenshot 2024-03-01 at 3 57 33 PM
Screenshot 2024-03-01 at 3 57 45 PM

File: internal/serve/validators/disbursement_instructions_validator.go [hunk 1]
```diff
@@ -65,3 +65,17 @@ func (iv *DisbursementInstructionsValidator) ValidateInstruction(instruction *da
...
+
+func (iv *DisbursementInstructionsValidator) SanitizeInstruction(instruction *data.DisbursementInstruction) *data.DisbursementInstruction {
+	var sanitizedInstruction data.DisbursementInstruction
+	sanitizedInstruction.Phone = strings.TrimSpace(instruction.Phone)
+	sanitizedInstruction.ID = strings.TrimSpace(instruction.ID)
+	sanitizedInstruction.Amount = strings.TrimSpace(instruction.Amount)
+	sanitizedInstruction.VerificationValue = strings.TrimSpace(instruction.VerificationValue)
+
+	if instruction.ExternalPaymentId != nil {
+		externalPaymentId := strings.TrimSpace(*instruction.ExternalPaymentId)
```
Comment (marcelosalloum) [inline internal/serve/validators/disbursement_instructions_validator.go:77]:
```suggestion
		externalPaymentID := strings.TrimSpace(*instruction.ExternalPaymentId)
```

File: internal/serve/httphandler/disbursement_handler_test.go [hunk 2]
```diff
@@ -854,6 +854,17 @@ func Test_DisbursementHandler_PostDisbursementInstructions(t *testing.T) {
...
+			name:           "valid input",
+			disbursementID: draftDisbursement.ID,
+			csvRecords: [][]string{
+				{"phone", "id", "amount", "verification"},
+				{"+380445555555", "123456789", "100.5", "1990-01-01"},
+				{"+380445555555", "123456789", "100.5", "1990-01-01"},
+			},
+			expectedStatus:  http.StatusOK,
+			expectedMessage: "File uploaded successfully",
+		},
```
Comment (marcelosalloum) [inline internal/serve/httphandler/disbursement_handler_test.go:867]: Is the idea here to validate that you can have multiple payments to the same receiver? If so, can you update the test name to make it clear?

**Review by marcelosalloum — APPROVED** [review]
LGTM! Plz, feel free to merge it after addressing the comments below.

File: internal/serve/httphandler/disbursement_handler_test.go (hunk 2 above)
Comment (marwen-abid) [reply internal/serve/httphandler/disbursement_handler_test.go]: No Sorry, I was testing something else there and forgot to remove this test. I'm removing it now.

**Review by marcelosalloum — COMMENTED** [review]
Hey @marwen-abid, will you bump the version and update the changelog in another PR? I believe it's more straightforward to do that in this one, right?

**Pull request merged**